### PR TITLE
SwiftVectorClass

### PR DESCRIPTION
**Pull request by mwflickner**
added vector class that matches python, erlang, lua string format, uses doubles

### CODE REVIEW

File: swift/vectors.swift [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+func + (left: Vector, right: Vector) -> Vector {
+    return Vector(x: left.x + right.x, y: left.y + right.y)
+}
+
+func * (left: Vector, right: Vector) -> Double {
+    return left.x * right.x + left.y * right.y
+}
+
+let u = Vector(x: 3,y: 4)
+let v = Vector(x: -5,y: 10)
```
Comment (jadengore) [inline swift/vectors.swift]: @mwflickner Perhaps some spaces for these parameters?

Comment (jadengore) [inline swift/vectors.swift]: Also sorry for randomly coming in and reviewing your code :wink:

Comment (mwflickner) [inline swift/vectors.swift]: lol its okay. I'm bored at JFK right now so it's keeping me entertained. just threw up some fixed stuff

**Comment by mwflickner** [comment]
Here's a good read on this stuff if anyone is interested.
https://developer.apple.com/library/ios/documentation/Swift/Conceptual/Swift_Programming_Language/AdvancedOperators.html

**Comment by jadengore** [comment]
@mwflickner LGTM, didn't know Swift had operator overloading. Looks super awesome!

**Comment by rtoal** [comment]
That link is awesome @mwflickner

**Comment by rtoal** [comment]
Nicely done, thanks

**Pull request merged**